src/test/integration/IntegrationDeployer.t.sol

```diff
@@ -0,0 +1,489 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+// Imports
+import "@openzeppelin/contracts/token/ERC20/presets/ERC20PresetFixedSupply.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin/contracts/proxy/beacon/IBeacon.sol";
+import "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+import "forge-std/Test.sol";
+
+import "src/contracts/core/DelegationManager.sol";
+import "src/contracts/core/StrategyManager.sol";
+import "src/contracts/core/Slasher.sol";
+import "src/contracts/strategies/StrategyBase.sol";
+import "src/contracts/pods/EigenPodManager.sol";
+import "src/contracts/pods/EigenPod.sol";
+import "src/contracts/pods/DelayedWithdrawalRouter.sol";
+import "src/contracts/permissions/PauserRegistry.sol";
+
+import "src/test/mocks/EmptyContract.sol";
+import "src/test/mocks/ETHDepositMock.sol";
+import "src/test/mocks/BeaconChainOracleMock.sol";
+
+import "src/test/integration/users/User.sol";
+
+abstract contract IntegrationDeployer is Test {
+
+    Vm cheats = Vm(HEVM_ADDRESS);
+
+    // Core contracts to deploy
+    DelegationManager delegationManager;
+    StrategyManager strategyManager;
+    EigenPodManager eigenPodManager;
+    PauserRegistry pauserRegistry;
+    Slasher slasher;
+    IBeacon eigenPodBeacon;
+    EigenPod pod;
+    DelayedWithdrawalRouter delayedWithdrawalRouter;
+
+    // Base strategy implementation in case we want to create more strategies later
+    StrategyBase baseStrategyImplementation;
+
+    // List of LST strategies
+    IStrategy[] _strategies;
+
+    // TODO - rename to time machine??
+    Global global;
```

Comment:
yes, something like `TimeMachine` or even just `Snapshot` would be clearer

Reply:
i like TimeMachine. i'm trying really hard not to leave all kinds of jokes all over the place

Reply:
but some stuff can be fun. i like having fun when i write tests

Reply:
[6ef08a6](https://github.com/Layr-Labs/eigenlayer-contracts/pull/335/commits/6ef08a676fe70389631c4627a9aa2d825a660884)